### PR TITLE
Fix for issue #57

### PR DESCRIPTION
**Pull request by wspowell**
The problem described in issue #57 occurs because the object is removed on the remove() call then is not present during the update() call. This is because the narrow phase attempts to resolve colliding objects through the use of the ContactSignal. However, because the object was previously removed, one object in the pair in the callback handler is essentially null. The solution I made was to store a list of objects that are to be removed, but not actually do the removing until the end of update().

Basically, mark the object to be removed (but not remove) in remove(), then after update() remove all that have been marked.

### CODE REVIEW

**Comment by sebcrozet** [comment]
Your analysis of the problem is correct. Though the fix goes deeper than this. The `.deferred_remove(...)` method from the broad phase won't actually remove the object until the next update, so performing the removal that way will simply delay the crash (simply add another `.update()` to your test to make it fail). I'll used your work as a basis for a proper fix.

Thanks!

**Comment by sebcrozet** [comment]
Closing in favor of #114 .
